### DISCUSSION TITLE
How to create a global function that can access over full adonis app and also edge file.

### DISCUSSION THREAD

**Discussion by emrancu** [adonisjs/core#4587] [Help] [answered]
I need to call a function of class's method in edge component.

**Comment by aikrom** [comment]
[Look here](https://github.com/adonisjs/core/blob/develop/providers/edge_provider.ts)

**Reply by emrancu** [reply, marked as answer]
## 1. Create a ts file and add to preload on ```adonisrc.ts``` on root 
```javascript
  preloads: [() => import('path'))
```

### In your custom file:
```javascript
import edge from 'edge.js'
```

and set like:

```javascript

const getData = ()=> return 'ok'

edge.global('getData', getData)

```

Now You can Access from .edge file like:

```
 {{ getData() }}
```
